### PR TITLE
Allow shuffle when auto-batching disabled in DataLoader

### LINKED ISSUE #35761: Allow shuffle when auto-batching disabled in `DataLoader`
## 🚀 Feature
Allow `DataLoader(batch_size=None, shuffle=True)`

## Motivation

My model handles one sample a time. So, I don't need auto-batching, but I want my dataset to be shuffled. However, when I use `DataLoader(batch_size=None, shuffle=True)`, it raises `batch_size=None option disables auto-batching and is mutually exclusive with shuffle, and drop_last`.

But I can't see why they are mutually exclusive.

## Pitch

Allow `DataLoader(batch_size=None, shuffle=True)`.

Basically, just remove the `shuffle` check from this line:  https://github.com/pytorch/pytorch/blob/ada647214f3c39b3fdf624dd62f8295c92144617/torch/utils/data/dataloader.py#L202

## Alternatives

I can:
* Use `batch_size=1` then remove the batching dimension afterwards
* Use `sampler=RandomSampler(dataset)`

But these makes me feel like I'm fighting with DataLoader. Why I cannot just use `shuffle=True`?

## Additional context

cc @SsnL

### PR DESCRIPTION
**Pull request by ydaiming**
Fix #35761
cc @SsnL

Note: closed the other PR for this new branch.

### CODE REVIEW

**Review by ssnl — APPROVED** [review]
LGTM
